File: WebSite with CosmosDB/server/Basketball/Microservice/Models/Season.cs
using Microsoft.Azure.Cosmos.Spatial;
using NBA.Infrastructure;

namespace NBA.Models
{
    public class Season : BasketballModel
    {
        public List<TeamScalation> Teams { get; set; } = [];

        public override (bool Success, string Message) Validate()
        {
            //To be improved with FluentValidation library
            if (Id is null)
                return (false, "id cannot be null");

            if (Teams is null)
                return (false, "teams cannot be null");

            foreach (var team in Teams)
            {
                var (Success, Message) = team.Validate();
                if (!Success)
                    return (false, $"team of id '{team.Id}' is invalid due to '{Message}'");
            }

            return (true, null);
        }
    }
}

File: Vue-Nuxt with Java-MongoDB/server/Basketball/Microservice/src/main/java/com/nba/microservice/models/Season.java
package com.nba.microservice.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.nba.microservice.infrastructure.BasketballModel;
import com.nba.microservice.infrastructure.ValidationResult;
import com.nba.microservice.models.ValueObjects.TeamScalation;
import java.util.List;

public class Season extends BasketballModel {
    @JsonProperty("teams")
    private List<TeamScalation> teams;

    public Season() {
    }

    public Season(String id, List<TeamScalation> teams) {
        this.setId(id);
        this.teams = teams;
    }

    public List<TeamScalation> getTeams() {
        return teams;
    }

    public void setTeams(List<TeamScalation> teams) {
        this.teams = teams;
    }

    @Override
    public ValidationResult validate() {
        if (getId() == null) {
            return new ValidationResult(false, "id cannot be null");
        }

        if (teams == null) {
            return new ValidationResult(false, "teams cannot be null");
        }

        for (TeamScalation team : teams) {
            ValidationResult result = team.validate();
            if (!result.isSuccess()) {
                return new ValidationResult(false,
                        "team of id '" + team.getId() + "' is invalid due to '" + result.getMessage() + "'");
            }
        }

        return new ValidationResult(true, null);
    }
}